include/rcutils/get_env.h

```diff
@@ -61,6 +61,30 @@ RCUTILS_WARN_UNUSED
 const char *
 rcutils_get_env(const char * env_name, const char ** env_value);
 
+// Retrieve the full path to the home directory.
+/*
+ * The c-string which is returned is only valid until the next time this
+ * function is called, because it is a direct pointer to the static storage.
+ * Also note that the string returned here should *not* be freed.
+ *
+ * The function work by first trying to get the contents of the HOME environment
+ * variable.  If that variable exists and is non-empty, that will be returned.
+ * Otherwise, the function will try to get the contents of the USERPROFILE
+ * environment variable.  If that variable exists and is non-empty, that will be
+ * returned.  If neither exists, NULL will be returned.  The above algorithm
+ * should be portable across both Unix and Windows.
+ *
+ * The home directory will be truncated at 2048 characters on Windows.
+ *
+ * This function is not thread-safe.
+ *
+ * \return The home directory on success, NULL on failure.
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
```

Comment:
I don't think this macro is necessary here. The return value is nothing the user needs to check for error handling.

Reply:
`rcutils_get_home_dir` can actually fail (if no HOME or USERPROFILE exists for some reason).  So I think reminding the user to check is probably a good thing.

`rcutils_get_pid` cannot fail, but on the other hand, there is no reason to call it if you aren't going to use the result.  Thus, marking it WARN_UNUSED is a reminder to the user that they probably have a bug in their program.